generic.json

```diff
@@ -8,25 +8,19 @@
         },
         "oneOf": [
```

Comment:
Actually I think the problem comes from here. I didn't find a way to include a `oneOf` construct *directly*  inside the `properties` field. This result in an invalid schema

Try this simple schema on [an online validator](https://jsonschemalint.com/#!/version/draft-07/markup/json)
```json
{
  "type": "object",
  "properties": {
    "oneOf": [
      {
        "jsonrpc": {
          "type": "string"
        }
      },
      {
        "m": {
          "type": "integer"
        }
      }
    ]
  }
}
```

Reply:
I don't know, the code below passes the validator and the only difference is adding "definitions". I think it has to do with type: object and oneOf not being interpreted as an object (when properties is the object and its type is free, seems to work better).

Otherwise, what you did with only using the xxxOf on the required fields might be the alternative?

```
{
  "type": "object",
      "definitions": {
           "properties": {
              "oneOf": [
             {
              "jsonrpc": {
               "type": "string"
             }
         },
       {
        "m": {
          "type": "integer"
        }
      }
    ]
  }
}
```

Reply:
Well, it *is* a valid JsonSchema (if you add a missing `}` at the end), but it doesn't check what we want. Your schema matchs an `object` (line 2) but doesn't know what to do with the `definitions`. Thus, this schema will match ANY JsonObject (since by default `"required": []` and `additionalProperties: true`). To convince you, try adding the `additionalProperties: false` field at the end like this:

```json
{
	"type": "object",
	"definitions": {
		"properties": {
			"oneOf": [
				{ "jsonrpc": { "type": "string" } },
				{ "m": { "type": "integer" } }
			]
		}
	},
	"additionalProperties": false
}
```

Now, this code accepts ONLY the empty JsonObject (since again, it doesnt know what to do with definitions and is now restricted to the properties specified (namely, none))

Reply:
As I read this code, what I understand (conceptually) is that you're willing to accept either multiple methods or multiple return. 

I can give you a helping hand this afternoon, but please clarify for me what you're trying to achieve and what's blocking.  Let's start with an English description, and we'll transform that to JSON Schema in a second step :) 

In the mean time, have a look at this:
https://json-schema.org/understanding-json-schema/reference/combining.html

If need be (not sure you need to know this), there are also conditionals:
https://json-schema.org/understanding-json-schema/reference/conditionals.html

Reply:
Thanks! In English, the generic file should accept either a request or a return. When it accepts a request it should accept either of the 5 methods. When it accepts a return, it should accept either the basic return or the catchup return (which has more content than the basic return).

Lower in the structure, but this is not handled in the generic file, a basic return should accept either a successful return or an error return, and some methods like publish should accept either of the many options for the data fields (createLAO, etc.)

Either can be represented by either (^^) oneOf or anyOf. When differentiating between request and return, the fields are so different than using oneOf is safe. When differentiating between the various data of publish, as some might have similar fields, anyOf is the better choice.

Reply:
I'll come back on this tomorrow morning first thing, did not have time this afternoon :-( Sorry.